field_friend/interface/components/support_point_dialog.py

```diff
@@ -0,0 +1,88 @@
+from typing import TYPE_CHECKING, Callable
+from uuid import uuid4
+
+import rosys
+from nicegui import ui
+
+from field_friend.automations.field import RowSupportPoint
+from field_friend.interface.components.monitoring import CameraPosition
+from field_friend.localization import GeoPoint
+import shapely
+if TYPE_CHECKING:
+    from field_friend.system import System
+
+
+class SupportPointDialog:
+
+    def __init__(self, system: 'System'):
+        self.front_cam = next((value for key, value in system.mjpeg_camera_provider.cameras.items()
+                               if CameraPosition.FRONT in key), None) if hasattr(system, 'mjpeg_camera_provider') else None
+        self.steerer = system.steerer
+        self.gnss = system.gnss
+        self.field_provider = system.field_provider
+        self.row_name: int = 1
+        self.support_point_coordinates: GeoPoint | None = None
+        self.next: Callable = self.find_support_point
+
+        with ui.dialog() as self.dialog, ui.card().style('width: 900px; max-width: none'):
+            with ui.row().classes('w-full no-wrap no-gap'):
+                self.row_sight = ui.interactive_image().classes('w-3/5')
+                with ui.column().classes('items-center  w-2/5 p-8'):
+                    self.headline = ui.label().classes('text-lg font-bold')
+                    self.content = ui.column().classes('items-center')
+                    # NOTE: the next function is replaced, hence we need the lambda
+                    ui.button('Next', on_click=lambda: self.next())
+        ui.timer(0.1, self.update_front_cam)
+        self.open()
+
+    def open(self) -> None:
+        self.next()
+        self.dialog.open()
+
+    def find_support_point(self) -> None:
+        self.headline.text = 'Drive to Row'
+        self.row_sight.content = '<line x1="50%" y1="0" x2="50%" y2="100%" stroke="#6E93D6" stroke-width="6"/>'
+        with self.content:
+            rosys.driving.joystick(self.steerer, size=50, color='#6E93D6')
+            ui.label('1. Drive the robot on the row you want to give a fixed support point.').classes(
+                'text-lg')
+            ui.label('2. Enter the row number for the support point:').classes('text-lg')
+            ui.number(
+                label='Row Number', min=1, max=self.field_provider.fields[0].row_number - 1 if self.field_provider.fields else 1, step=1, value=1) \
```

Comment:
```suggestion
                label='Row Number', min=1, max=self.field_provider.fields[0].row_number if self.field_provider.fields else 1, step=1, value=1) \
```
I think -1 is wrong because we use human numbers here, not the index